main.js

```diff
@@ -1,123 +1,169 @@
+
+    // First step: record the hash on an anchoring channel (IOTA Stream)
+    const channelDetails = await sendToIOTA(action, target);
+
+    // Second step: update the target custom fields with the channel details
+    const updatedTarget = await updateTarget(target, targetType, channelDetails);
+
+    // Final step: Send the confirmation action
+    const confirmationAction = await createConfirmation(action, updatedTarget, targetType);
+
+    logger.info(`Confirmation action: ${confirmationAction.id}`);
+  });
+
+/**
+ * Read the complete target object from the action.
+ * This is either a Thng, product, or collection.
+ *
+ * @returns target that contains all data about the target and the target type
+ */
+async function readTarget (action) {
+  let targetType;
+
+  if (action.thng) {
+    targetType = 'thng';
+  } else if (action.product) {
+    targetType = 'product';
+  } else if (action.collection) {
+    targetType = 'collection';
+  } else {
+    throw new Error('No target was specified!');
+  }
+
+  const target = await app[targetType](action[targetType]).read();
+
+  return { target, targetType };
+}
 
 /**
  * Send the action's SHA 256 hash to IOTA.
  *
- * @returns {Promise} Promise that resolves the message root, and the new MAM state.
+ * @param {object} action The concerned action
+ * @param {object} target The target on which the action is executed
+ *
+ * @returns {object} the channel details
  */
-const sendToIOTA = () => {
+async function sendToIOTA (action, target) {
   // Hash the action with SHA256
-  const sha256 = hashJs.sha256()
-    .update(jsonSortify(action))
-    .digest('hex');
+  const sha256 = hashJs.sha256().update(jsonSortify(action)).digest('hex');
   logger.debug(`Action SHA256: ${sha256}`);
 
-  // Use pre-recorded MAM state, else use a new one
-  let mamState = Mam.changeMode(Mam.init(NODE_ADDRESS), 'public');
-  if(target.customFields && target.customFields.iotaMamState) {
-    mamState = target.customFields.iotaMamState;
+  // We need to check whether a new channel is needed to be created or not
+  let channelDetails =
+    target.customFields && target.customFields.iotaAnchoringChannel;
+  let channel;
+  // The anchorage to be used to anchor this message (hash)
+  let anchorageID;
+
+  // Any Node connection params should be under the options object
+  let options;
+  if (NODE_ADDRESS) {
+    options = {
+      node: NODE_ADDRESS
+    };
   }
 
-  // Encode the payload
-  try {
-    const { root, payload, address } = Mam.create(mamState, asciiToTrytes(sha256));
-    return Mam.attach(payload, address, DEPTH, MWM).then(() => {
-      logger.debug(`root: ${root}`);
-      return { root, mamState };
-    }).catch((e) => {
-      // e is a string
-      throw new Error(e);
-    });
-  } catch (e) {
-    return Promise.reject('Failed to create and/or attach: ' + e.message + ' - ' + e.stack);
+  if (channelDetails) {
+    const channelID = channelDetails.channelID;
```

Comment:
These can be declared using destructuring syntax:
```js
const { channelID, seed } = channelDetails;
```

Reply:
👍